src/graph/planner/plan/Query.h

```diff
@@ -663,14 +719,20 @@ class Limit final : public SingleInputNode {
   Limit(QueryContext* qctx, PlanNode* input, int64_t offset, int64_t count)
       : SingleInputNode(qctx, Kind::kLimit, input) {
     offset_ = offset;
+    count_ = ConstantExpression::make(qctx_->objPool(), count);
+  }
+
+  Limit(QueryContext* qctx, PlanNode* input, int64_t offset, Expression* count)
+      : SingleInputNode(qctx, Kind::kLimit, input) {
+    offset_ = offset;
     count_ = count;
   }
 
   void cloneMembers(const Limit&);
 
  private:
   int64_t offset_{-1};
```

Comment:
Need change `offset` into an expression?

Reply:
offset won't push down

Reply:
It is better to ensure consistency in use.
```
go from ....| offset abs(1)+3 limit abs(3)+1
go from ....| limit abs(3)+1,abs(3)
```

Reply:
This is used for new syntax for `go ... limit [..]`